File: Microverse Challenges/module 3/3.4 towers-of-hanoi---part-1-Trast00/challenge.js
module.exports = function move(starting, goal) {
  // write your code here
  let used = 1
  while (starting === used || goal === used) {
    used +=1;
    if(used===4){used =  1}
  }

  
  let result = ""
  result += `${starting}->${used} `
  result += `${starting}->${goal} `
  result += `${used}->${goal}`
  console.log(result)
  return result
}

//console.log(move(1, 3))
// => 1->2 1->3 2->3

//console.log(move(2, 3))
// => 2->1 2->3 1->3

File: Microverse Challenges/module 5/5.1 insertion-sort-intro-Trast00/challenge.rb
def insertion_sort(array)
  # write your code here

  index = array.length-1
  shifted_element = array[index]
  result = []
  until array[index-1] <= shifted_element do # until previousElement is smaller than X
    
    array[index] = array[index-1]
    print "#{array.join(' ')}\n"
    index -=1
    array.unshift(shifted_element) if index == 0
    exit if index == 0
  end
  if index!=0  
    array[index]= shifted_element
    result << array
    print "#{array.join(' ')}\n"
  end
  #puts result
end

insertion_sort([1, 3, 2])
# => 1 3 3
#    1 2 3

puts '-' * 20

insertion_sort([1, 4, 6, 9, 3])
# => 1 4 6 9 9
#    1 4 6 6 9
#    1 4 4 6 9
#    1 3 4 6 9
